### ISSUE TITLE
Training annotations for community model

### ISSUE THREAD

**Issue by bw4sz** [sizhuoli/TreeCountSegHeight#5]
I am the maintainer of an open source python package for individual tree segmentation in RGB imagery. 

https://github.com/weecology/DeepForest

I am getting ready to retrain the base model with thousands of new annotations we've collected through research projects around the world. 

https://github.com/weecology/DeepForest/issues/340

I'd like to include the data used to train this model

In the paper

```
We manually delineated a total of 24,466 individual tree crowns from sampling plots of varying sizes distributed over Denmark, among which 19,771 crowns (49% in dense deciduous forests, 30% in dense coniferous forests, and 21% in nonforest areas) were assigned to the training data set, 2,016 crowns (46% in dense deciduous forests, 38% in dense coniferous forests, and 15% in nonforest areas) to the validation data set, and 2,679 crowns (48% in dense deciduous forests, 32% in dense coniferous forests, and 20% in nonforest areas) to the final test data set. The manual delineation took ∼3 weeks and involved creating annotations covering a large variety of tree species and landscape types.
```

I know that annotating crowns is very difficult work and I think its important that the community gets the maximum benefit out of this hard task. Thanks for your consideration.

Ben Weinstein

**Comment by robodrome** [comment]
I have identified a source that could also be helpful for training and testing models. Some municipalities in the Netherlands have open data on trees. We are talking about hundreds of thousands labeled trees, including age, size, species. No tree crowns however. Here's a [link to a website](https://openbomenkaart.org/alle_kaarten.html) that sums up a number of them. These are mostly urbanized areas, not dense forests.

In combination with RGB, CIR (can be converted to NDVI), DSM, DTM and pointcloud [open data](https://geotiles.nl/) this could be potentially be made into a useful dataset for training.

Hope it helps.

Best,
David

**Comment by bw4sz** [comment]
@robodrome Do you have any experience with this dataset? I see some geolocated trees

<img width="1693" alt="image" src="https://github.com/sizhuoli/TreeCountSegHeight/assets/1208492/57fd60f6-1ff1-4ed1-bdf0-aad0cd9340ff">

but no corresponding overhead imagery. Is that right?

**Comment by robodrome** [comment]
@bw4sz 
No I don't have experience with the Dutch tree maps. Yet. I am just starting to look into it. I found the paper and this code repo and I would like to test it on Dutch open data.

Bomenkaart.org is not an official source. Official sources are e.g. municipalities. Take Utrecht, which has an [ARCGIS viewer](https://gu-geo.maps.arcgis.com/apps/webappviewer/index.html?id=fc5af2aed5244703857e4e602b478178) and a [ARCGIS feature service](https://gu-geo.maps.arcgis.com/home/item.html?id=6ebc447768614159989abd454b660b7e). Or [Amsterdam bomenkaart](https://maps.amsterdam.nl/bomen/). The magic word in google is `bomenkaart`.

The link I provided (https://geotiles.nl/) directs you to a site where you can download hires aerial imagery for the Netherlands; RGB, CIR but also pointcloud data and DTM, DSM. The latter have 0.5m resolution, but I have used the AHN4 pointcloud to create 0.25m DTM, DSM and CHM.

Geotiles is a convenient platform to get the data. I think it is funded by TU Delft (University). [PDOK](https://www.pdok.nl/introductie/-/article/pdok-luchtfoto-rgb-open-) is the official source, which is funded by the Dutch government.

Does that answer your questions?

Best,
David

PS. Also, at bomenkaart.org you can select map type in the right bottom.

![image](https://github.com/sizhuoli/TreeCountSegHeight/assets/98538838/bac98665-75f5-477a-83f4-7e770454bd81)

**Comment by bw4sz** [comment]
Thanks for your help on this, there is a lot of data out there and each one requires abit of experience. Looking at any given dataset, is it possible to figure out which tile from (https://geotiles.nl/)  needs to be downloaded?

So if i'm in browser and I download this point data, or really all the point data.
<img width="1724" alt="image" src="https://github.com/sizhuoli/TreeCountSegHeight/assets/1208492/d8aecbdc-6c7a-4a13-8273-6b7180b14472">

I'll need to manually go to 
<img width="1724" alt="Screenshot 2023-11-07 at 7 05 24 AM" src="https://github.com/sizhuoli/TreeCountSegHeight/assets/1208492/0c5e78f7-c085-4ba1-a7d4-67d3026be852">

and figure out geographically which tiles they correspond to? Any way to wget all of them, I could search using python. Thanks for your thoughts, i appreciate it.

**Comment by bw4sz** [comment]
also @robodrome, we've hijacked this thread, you are welcome to email me benweinstein2010@gmail.com, sorry @sizhuoli, let me know if there is progress on getting the other data.

**Comment by sizhuoli** [comment]
Hi Ben @bw4sz, I wrote twice to the contact I know but unfortunately haven't got a reply yet:)

**Comment by robodrome** [comment]
> Thanks for your help on this, there is a lot of data out there and each one requires abit of experience. Looking at any given dataset, is it possible to figure out which tile from (https://geotiles.nl/) needs to be downloaded?
> 
> So if i'm in browser and I download this point data, or really all the point data. 
> I'll need to manually go to 
> 
> and figure out geographically which tiles they correspond to? Any way to wget all of them, I could search using python. Thanks for your thoughts, i appreciate it.

@bw4sz 

- Here's a nice blog on the Dutch grid used on geotiles: https://weblog.fwrite.org/kaartbladen/
- There are also some links to grid geopackages there. 
- You could use a GIS query on those geopackages
- Note that the Netherlands uses EPSG:28992 for extra accuracy.

It will be some work, but I think this data could be useful for training sets. It is likely very accurate and detailed.
